### PR TITLE
Added webform autocomplete element 

### LINKED ISSUE #825: Webform Autocomplete Functionality
- [x] Add autocomplete element plugin
- [x] Allow developers to access the Option menu view to add a predefined list

### PR DESCRIPTION
**Pull request by sarongebre**
Added Webform autocomplete element with support for predefined lists and YAML predefined list source configuration for developers. I've adjusted the permissions to only allow the Options tab and not other configurations. 

closes #825

### CODE REVIEW

File: web/modules/custom/unl_webform/unl_webform.permissions.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 administer unl webform download settings:
   title: 'Administer UNL Webform download settings'
   description: 'Allows a user to configure UNL Webform download key settings.'
+
+access webform options:
```
Comment (ericras) [inline web/modules/custom/unl_webform/unl_webform.permissions.yml:5]: Can this be changed to "access unl webform options" to make sure we never have a collision with the main webform module in the future?

**Review by ericras — CHANGES_REQUESTED** [review]

File: web/modules/custom/unl_webform/src/Routing/RouteSubscriber.php [hunk 2]
```diff
@@ -8,19 +8,34 @@
 /**
  * Listens to the dynamic route events.
  */
-class RouteSubscriber extends RouteSubscriberBase {
+class RouteSubscriber extends RouteSubscriberBase
+{
```
Comment (ericras) [inline web/modules/custom/unl_webform/src/Routing/RouteSubscriber.php:12]: You might need to set your IDE to not change formatting like this when you save the file

File: web/modules/custom/unl_webform/unl_webform.permissions.yml (hunk 1 above)
Comment (sarongebre) [reply web/modules/custom/unl_webform/unl_webform.permissions.yml:5]: Okay, I will change that

File: web/modules/custom/unl_webform/src/Routing/RouteSubscriber.php (hunk 2 above)
Comment (sarongebre) [reply web/modules/custom/unl_webform/src/Routing/RouteSubscriber.php:12]: Okay.

File: web/modules/custom/unl_webform/unl_webform.permissions.yml (hunk 1 above)
Comment (sarongebre) [reply web/modules/custom/unl_webform/unl_webform.permissions.yml]: I have made the change

File: web/modules/custom/unl_webform/src/Routing/RouteSubscriber.php (hunk 2 above)
Comment (sarongebre) [reply web/modules/custom/unl_webform/src/Routing/RouteSubscriber.php:12]: I have made the change

**Review by ericras — APPROVED** [review]

**Pull request merged**